src/main.zig

```diff
@@ -6772,6 +6775,7 @@ fn cmdFetch(
     var override_global_cache_dir: ?[]const u8 = try EnvVar.ZIG_GLOBAL_CACHE_DIR.get(arena);
     var debug_hash: bool = false;
     var save: union(enum) { no, yes, name: []const u8 } = .no;
+    var preserve_url: bool = false;
```

Comment:
This needs to be incorporated into the type of `save`. I think a better set of flags would be `--save-exact` and `--save-exact=[name]`.

Reply:
See #19941